### PR TITLE
throw SigningException as documented

### PR DESCRIPTION
**Pull request by elharo**
Least invasive change that partially addresses #314 but  we need to change the API to correctly fix this. @chingor13

### CODE REVIEW

File: oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java [hunk 1]
```diff
@@ -304,12 +304,15 @@ public String getAccount() {
...
-        transportFactory.create(),
-        toSign,
-        Collections.<String, Object>emptyMap());
+    try {
+      String account = getAccount();
+      return IamUtils.sign(
+          account, this, transportFactory.create(), toSign, Collections.<String, Object>emptyMap());
+    } catch (SigningException ex) {
+      throw ex;
+    } catch (RuntimeException ex) {
```
Comment (chingor13) [inline oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java]: Let's add a TODO to clean this up that mentions #314

Comment (elharo) [reply oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java]: done. Also added some JavaDoc to getAccount.

File: oauth2_http/javatests/com/google/auth/oauth2/ComputeEngineCredentialsTest.java [hunk 2]
```diff
@@ -296,7 +295,27 @@ public void sign_sameAs() throws IOException {
     ComputeEngineCredentials credentials =
         ComputeEngineCredentials.newBuilder().setHttpTransportFactory(transportFactory).build();
 
-    assertArrayEquals(expectedSignature, credentials.sign(expectedSignature));
+    assertEquals(defaultAccountEmail, credentials.getAccount());
```
Comment (chingor13) [inline oauth2_http/javatests/com/google/auth/oauth2/ComputeEngineCredentialsTest.java]: I think we want to still test the signature in this test. Perhaps keep both assertions or split the `getAccount()` test into its own test.

Comment (elharo) [reply oauth2_http/javatests/com/google/auth/oauth2/ComputeEngineCredentialsTest.java]: editing glitch. fixed.

**Pull request merged**